lib/bolt/node/winrm.rb

```diff
@@ -140,6 +144,183 @@ def shell_init
     }
   }
 }
+
+function Write-Stream {
```

Comment:
In a separate commit we might want to think about moving this bootstrapping code to its own file.  I had a feeling it was going to grow over time, and the inline string it starting to get a little unwieldy.

Reply:
We should, this is actually a copypasta from https://github.com/puppetlabs/pxp-agent/blob/409c82cd4ffa16a65e1c231b444619774e520cd9/exe/PowershellShim-Helper.ps1 which is used by pxp-agent. If we move this to a separate file it could potentially be shared.

Reply:
One more note here: I originally used:
```powershell
Add-Type -Assembly System.Web.Extensions
```
then I could use the `JavaScriptSerializer` class from that assembly for parsing the JSON like this:
```powershell
$_task_args = (New-Object System.Web.Script.Serialization.JavaScriptSerializer).DeserializeObject(
   $utf8.GetString([Convert]::FromBase64String('#{Base64.encode64(JSON.dump(arguments))}'))
)
```
which worked on my development machine, but not on Windows Nano VM spun up by vagrant.
If we could use that class we could drop all of the code added here.
Could we declare the assembly as a requirement?

Reply:
I don't think we can guarantee `System.Web.Extensions` is available on all target machines / PowerShell / .NET Framework versions unfortunately.

Reply:
oh well ...